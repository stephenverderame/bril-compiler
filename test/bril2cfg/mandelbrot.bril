@main {
  v12: int = const 1;
  space: int = id v12;
  v13: int = const 8;
  value: int = id v13;
  v14: float = const 0;
  v15: float = const 1;
  v16: float = fsub v14 v15;
  neg_1: float = id v16;
  v17: float = const 0;
  v18: float = const 2;
  v19: float = fsub v17 v18;
  neg_2: float = id v19;
  v21: float = const 1;
  j: float = id v21;
.for.cond.20:
  v22: float = id j;
  v23: float = id neg_1;
  v24: bool = fge v22 v23;
  br v24 .for.body.20 .for.end.20;
.for.body.20:
  v25: int = const 0;
  num: int = id v25;
  v26: int = const 0;
  cnt: int = id v26;
  v28: float = id neg_2;
  i: float = id v28;
.for.cond.27:
  v29: float = id i;
  v30: float = const 0.5;
  v31: bool = fle v29 v30;
  br v31 .for.body.27 .for.end.27;
.for.body.27:
  v32: float = id i;
  v33: float = id j;
  v34: float = const 0;
  v35: float = const 0;
  v36: int = const 0;
  x: int = call @f v32 v33 v34 v35 v36;
  x: int = id x;
  v38: int = id x;
  v39: int = const 50;
  v40: bool = lt v38 v39;
  br v40 .then.37 .else.37;
.then.37:
  v41: int = id space;
  v42: int = id cnt;
  v: int = call @pow10 v41 v42;
  v: int = id v;
  v43: int = id num;
  v44: int = id v;
  v45: int = add v43 v44;
  num: int = id v45;
  jmp .endif.37;
.else.37:
  v46: int = id value;
  v47: int = id cnt;
  v: int = call @pow10 v46 v47;
  v: int = id v;
  v48: int = id num;
  v49: int = id v;
  v50: int = add v48 v49;
  num: int = id v50;
.endif.37:
  v51: int = id cnt;
  v52: int = const 1;
  v53: int = add v51 v52;
  cnt: int = id v53;
  v54: float = id i;
  v55: float = const 0.225;
  v56: float = fadd v54 v55;
  i: float = id v56;
  jmp .for.cond.27;
.for.end.27:
  v57: int = id num;
  print v57;
  v58: int = const 0;
  v59: float = id j;
  v60: float = const 0.015;
  v61: float = fsub v59 v60;
  j: float = id v61;
  jmp .for.cond.20;
.for.end.20:
}
@f(X: float, Y: float, x: float, y: float, n: int): int {
  v0: float = id x;
  v1: float = id x;
  v2: float = fmul v0 v1;
  v3: float = id y;
  v4: float = id y;
  v5: float = fmul v3 v4;
  v6: float = fadd v2 v5;
  r: float = id v6;
  v8: float = id r;
  v9: float = const 4;
  v10: bool = flt v8 v9;
  br v10 .then.7 .else.7;
.then.7:
  v12: int = id n;
  v13: int = const 100;
  v14: bool = lt v12 v13;
  br v14 .then.11 .else.11;
.then.11:
  v15: float = id x;
  v16: float = id x;
  v17: float = fmul v15 v16;
  v18: float = id y;
  v19: float = id y;
  v20: float = fmul v18 v19;
  v21: float = fsub v17 v20;
  v22: float = id X;
  v23: float = fadd v21 v22;
  x2: float = id v23;
  v24: float = const 2;
  v25: float = id x;
  v26: float = fmul v24 v25;
  v27: float = id y;
  v28: float = fmul v26 v27;
  v29: float = id Y;
  v30: float = fadd v28 v29;
  y2: float = id v30;
  v31: int = const 1;
  v32: float = id X;
  v33: float = id Y;
  v34: float = id x2;
  v35: float = id y2;
  v36: int = id n;
  v37: int = const 1;
  v38: int = add v36 v37;
  v39: int = call @f v32 v33 v34 v35 v38;
  v40: int = add v31 v39;
  fv: int = id v40;
  v41: int = id fv;
  ret v41;
  jmp .endif.11;
.else.11:
.endif.11:
  jmp .endif.7;
.else.7:
.endif.7:
  v42: int = const 0;
  ret v42;
}
@pow10(n: int, e: int): int {
  v1: int = id e;
  i: int = id v1;
.for.cond.0:
  v2: int = id i;
  v3: int = const 0;
  v4: bool = gt v2 v3;
  br v4 .for.body.0 .for.end.0;
.for.body.0:
  v5: int = id n;
  v6: int = const 10;
  v7: int = mul v5 v6;
  n: int = id v7;
  v8: int = id i;
  v9: int = const 1;
  v10: int = sub v8 v9;
  i: int = id v10;
  jmp .for.cond.0;
.for.end.0:
  v11: int = id n;
  ret v11;
}
