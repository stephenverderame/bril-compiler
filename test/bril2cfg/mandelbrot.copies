digraph {
	subgraph cluster_main {
		label="main()";
		rankdir="TB";
		main_0 [label="START", shape="rectangle", style="rounded"];
		main_2 [label="[]\n----------\nv12 := 1\nspace := v12\nv13 := 8\nvalue := v13\nv14 := 0\nv15 := 1\nv16 := v14 - v15\nneg_1 := v16\nv17 := 0\nv18 := 2\nv19 := v17 - v18\nneg_2 := v19\nv21 := 1\nj := v21\n----------\n[j = v21, neg_1 = v16, neg_2 = v19, space = v12, value = v13]", shape="rectangle", style="rounded"];
		main_3 [label="[neg_1 = v16, neg_2 = v19, space = v12, v = v, v23 = neg_1, v28 = neg_2, v29 = i, v38 = x, v41 = space, v44 = v, v46 = value, v49 = v, v57 = num, value = v13, x = x]\n----------\nv22 := j\nv23 := neg_1\nv24 := v22 >= v23\nif v24\n----------\n[neg_1 = v16, neg_2 = v19, space = v12, v = v, v22 = j, v23 = neg_1, v28 = neg_2, v29 = i, v38 = x, v41 = space, v44 = v, v46 = value, v49 = v, v57 = num, value = v13, x = x]", shape="rectangle", style="rounded"];
		main_4 [label="[neg_1 = v16, neg_2 = v19, space = v12, v = v, v22 = j, v23 = neg_1, v28 = neg_2, v29 = i, v38 = x, v41 = space, v44 = v, v46 = value, v49 = v, v57 = num, value = v13, x = x]\n----------\nv25 := 0\nnum := v25\nv26 := 0\ncnt := v26\nv28 := neg_2\ni := v28\n----------\n[cnt = v26, i = v28, neg_1 = v16, neg_2 = v19, num = v25, space = v12, v = v, v22 = j, v23 = neg_1, v28 = neg_2, v38 = x, v41 = space, v44 = v, v46 = value, v49 = v, value = v13, x = x]", shape="rectangle", style="rounded"];
		main_5 [label="[neg_1 = v16, neg_2 = v19, space = v12, v = v, v22 = j, v23 = neg_1, v28 = neg_2, v33 = j, v38 = x, v41 = space, v44 = v, v46 = value, v49 = v, value = v13, x = x]\n----------\nv29 := i\nv30 := 0.5\nv31 := v29 <= v30\nif v31\n----------\n[neg_1 = v16, neg_2 = v19, space = v12, v = v, v22 = j, v23 = neg_1, v28 = neg_2, v29 = i, v33 = j, v38 = x, v41 = space, v44 = v, v46 = value, v49 = v, value = v13, x = x]", shape="rectangle", style="rounded"];
		main_6 [label="[neg_1 = v16, neg_2 = v19, space = v12, v = v, v22 = j, v23 = neg_1, v28 = neg_2, v29 = i, v33 = j, v38 = x, v41 = space, v44 = v, v46 = value, v49 = v, value = v13, x = x]\n----------\nv32 := i\nv33 := j\nv34 := 0\nv35 := 0\nv36 := 0\nx := f(v32, v33, v34, v35, v36)\nx := x\nv38 := x\nv39 := 50\nv40 := v38 < v39\nif v40\n----------\n[neg_1 = v16, neg_2 = v19, space = v12, v = v, v22 = j, v23 = neg_1, v28 = neg_2, v29 = i, v32 = i, v33 = j, v38 = x, v41 = space, v44 = v, v46 = value, v49 = v, value = v13, x = x]", shape="rectangle", style="rounded"];
		main_7 [label="[neg_1 = v16, neg_2 = v19, space = v12, v = v, v22 = j, v23 = neg_1, v28 = neg_2, v29 = i, v32 = i, v33 = j, v38 = x, v41 = space, v44 = v, v46 = value, v49 = v, value = v13, x = x]\n----------\nv41 := space\nv42 := cnt\nv := pow10(v41, v42)\nv := v\nv43 := num\nv44 := v\nv45 := v43 + v44\nnum := v45\n----------\n[neg_1 = v16, neg_2 = v19, num = v45, space = v12, v = v, v22 = j, v23 = neg_1, v28 = neg_2, v29 = i, v32 = i, v33 = j, v38 = x, v41 = space, v42 = cnt, v44 = v, v46 = value, value = v13, x = x]", shape="rectangle", style="rounded"];
		main_8 [label="[neg_1 = v16, neg_2 = v19, space = v12, v = v, v22 = j, v23 = neg_1, v28 = neg_2, v29 = i, v32 = i, v33 = j, v38 = x, v41 = space, v44 = v, v46 = value, v49 = v, value = v13, x = x]\n----------\nv46 := value\nv47 := cnt\nv := pow10(v46, v47)\nv := v\nv48 := num\nv49 := v\nv50 := v48 + v49\nnum := v50\n----------\n[neg_1 = v16, neg_2 = v19, num = v50, space = v12, v = v, v22 = j, v23 = neg_1, v28 = neg_2, v29 = i, v32 = i, v33 = j, v38 = x, v41 = space, v46 = value, v47 = cnt, v49 = v, value = v13, x = x]", shape="rectangle", style="rounded"];
		main_9 [label="[neg_1 = v16, neg_2 = v19, space = v12, v = v, v22 = j, v23 = neg_1, v28 = neg_2, v29 = i, v32 = i, v33 = j, v38 = x, v41 = space, v42 = cnt, v44 = v, v46 = value, v47 = cnt, v49 = v, value = v13, x = x]\n----------\nv51 := cnt\nv52 := 1\nv53 := v51 + v52\ncnt := v53\nv54 := i\nv55 := 0.225\nv56 := v54 + v55\ni := v56\n----------\n[cnt = v53, i = v56, neg_1 = v16, neg_2 = v19, space = v12, v = v, v22 = j, v23 = neg_1, v28 = neg_2, v33 = j, v38 = x, v41 = space, v44 = v, v46 = value, v49 = v, value = v13, x = x]", shape="rectangle", style="rounded"];
		main_10 [label="[neg_1 = v16, neg_2 = v19, space = v12, v = v, v22 = j, v23 = neg_1, v28 = neg_2, v29 = i, v33 = j, v38 = x, v41 = space, v44 = v, v46 = value, v49 = v, value = v13, x = x]\n----------\nv57 := num\nprint v57\nv58 := 0\nv59 := j\nv60 := 0.015\nv61 := v59 - v60\nj := v61\n----------\n[j = v61, neg_1 = v16, neg_2 = v19, space = v12, v = v, v23 = neg_1, v28 = neg_2, v29 = i, v38 = x, v41 = space, v44 = v, v46 = value, v49 = v, v57 = num, value = v13, x = x]", shape="rectangle", style="rounded"];

		main_0 -> main_2;
		main_2 -> main_3;
		main_3 -> main_4 [label="T"];
		main_4 -> main_5;
		main_5 -> main_6 [label="T"];
		main_5 -> main_10 [label="F"];
		main_6 -> main_7 [label="T"];
		main_6 -> main_8 [label="F"];
		main_7 -> main_9;
		main_8 -> main_9;
		main_9 -> main_5;
		main_10 -> main_3;
	}
	subgraph cluster_f {
		label="f(X: float, Y: float, x: float, y: float, n: int)";
		rankdir="TB";
		f_0 [label="START", shape="rectangle", style="rounded"];
		f_2 [label="[]\n----------\nv0 := x\nv1 := x\nv2 := v0 * v1\nv3 := y\nv4 := y\nv5 := v3 * v4\nv6 := v2 + v5\nr := v6\nv8 := r\nv9 := 4\nv10 := v8 < v9\nif v10\n----------\n[r = v6, v0 = x, v1 = x, v3 = y, v4 = y, v8 = r]", shape="rectangle", style="rounded"];
		f_3 [label="[r = v6, v0 = x, v1 = x, v3 = y, v4 = y, v8 = r]\n----------\nv12 := n\nv13 := 100\nv14 := v12 < v13\nif v14\n----------\n[r = v6, v0 = x, v1 = x, v12 = n, v3 = y, v4 = y, v8 = r]", shape="rectangle", style="rounded"];
		f_4 [label="[r = v6, v0 = x, v1 = x, v12 = n, v3 = y, v4 = y, v8 = r]\n----------\nv15 := x\nv16 := x\nv17 := v15 * v16\nv18 := y\nv19 := y\nv20 := v18 * v19\nv21 := v17 - v20\nv22 := X\nv23 := v21 + v22\nx2 := v23\nv24 := 2\nv25 := x\nv26 := v24 * v25\nv27 := y\nv28 := v26 * v27\nv29 := Y\nv30 := v28 + v29\ny2 := v30\nv31 := 1\nv32 := X\nv33 := Y\nv34 := x2\nv35 := y2\nv36 := n\nv37 := 1\nv38 := v36 + v37\nv39 := f(v32, v33, v34, v35, v38)\nv40 := v31 + v39\nfv := v40\nv41 := fv\nreturn v41\n----------\n[fv = v40, r = v6, v0 = x, v1 = x, v12 = n, v15 = x, v16 = x, v18 = y, v19 = y, v22 = X, v25 = x, v27 = y, v29 = Y, v3 = y, v32 = X, v33 = Y, v34 = x2, v35 = y2, v36 = n, v4 = y, v41 = fv, v8 = r, x2 = v23, y2 = v30]", shape="rectangle", style="rounded"];
		f_7 [label="[r = v6, v0 = x, v1 = x, v12 = n, v3 = y, v4 = y, v8 = r]\n----------\nv42 := 0\nreturn v42\n----------\n[r = v6, v0 = x, v1 = x, v12 = n, v3 = y, v4 = y, v8 = r]", shape="rectangle", style="rounded"];

		f_0 -> f_2;
		f_2 -> f_3 [label="T"];
		f_2 -> f_7 [label="F"];
		f_3 -> f_4 [label="T"];
		f_3 -> f_7 [label="F"];
	}
	subgraph cluster_pow10 {
		label="pow10(n: int, e: int)";
		rankdir="TB";
		pow10_0 [label="START", shape="rectangle", style="rounded"];
		pow10_2 [label="[]\n----------\nv1 := e\ni := v1\n----------\n[i = v1, v1 = e]", shape="rectangle", style="rounded"];
		pow10_3 [label="[v1 = e]\n----------\nv2 := i\nv3 := 0\nv4 := v2 > v3\nif v4\n----------\n[v1 = e, v2 = i]", shape="rectangle", style="rounded"];
		pow10_4 [label="[v1 = e, v2 = i]\n----------\nv5 := n\nv6 := 10\nv7 := v5 * v6\nn := v7\nv8 := i\nv9 := 1\nv10 := v8 - v9\ni := v10\n----------\n[i = v10, n = v7, v1 = e]", shape="rectangle", style="rounded"];
		pow10_5 [label="[v1 = e, v2 = i]\n----------\nv11 := n\nreturn v11\n----------\n[v1 = e, v11 = n, v2 = i]", shape="rectangle", style="rounded"];

		pow10_0 -> pow10_2;
		pow10_2 -> pow10_3;
		pow10_3 -> pow10_4 [label="T"];
		pow10_3 -> pow10_5 [label="F"];
		pow10_4 -> pow10_3;
	}
}
