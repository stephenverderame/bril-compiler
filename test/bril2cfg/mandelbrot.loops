digraph {
	subgraph cluster_main {
		label="main()";
		rankdir="TB";
		main_0 [label="START", shape="rectangle", style="rounded"];
		main_2 [label="v12 := 1\nspace := v12\nv13 := 8\nvalue := v13\nv14 := 0\nv15 := 1\nv16 := v14 - v15\nneg_1 := v16\nv17 := 0\nv18 := 2\nv19 := v17 - v18\nneg_2 := v19\nv21 := 1\nj := v21", shape="rectangle", style="rounded"];
		main_3 [label="v22 := j\nv23 := neg_1\nv24 := v22 >= v23\nif v24", shape="rectangle", style="rounded"];
		main_4 [label="v25 := 0\nnum := v25\nv26 := 0\ncnt := v26\nv28 := neg_2\ni := v28", shape="rectangle", style="rounded"];
		main_5 [label="v29 := i\nv30 := 0.5\nv31 := v29 <= v30\nif v31", shape="rectangle", style="rounded"];
		main_6 [label="v32 := i\nv33 := j\nv34 := 0\nv35 := 0\nv36 := 0\nx := f(v32, v33, v34, v35, v36)\nx := x\nv38 := x\nv39 := 50\nv40 := v38 < v39\nif v40", shape="rectangle", style="rounded"];
		main_7 [label="v41 := space\nv42 := cnt\nv := pow10(v41, v42)\nv := v\nv43 := num\nv44 := v\nv45 := v43 + v44\nnum := v45", shape="rectangle", style="rounded"];
		main_8 [label="v46 := value\nv47 := cnt\nv := pow10(v46, v47)\nv := v\nv48 := num\nv49 := v\nv50 := v48 + v49\nnum := v50", shape="rectangle", style="rounded"];
		main_9 [label="v51 := cnt\nv52 := 1\nv53 := v51 + v52\ncnt := v53\nv54 := i\nv55 := 0.225\nv56 := v54 + v55\ni := v56", shape="rectangle", style="rounded"];
		main_10 [label="v57 := num\nprint v57\nv58 := 0\nv59 := j\nv60 := 0.015\nv61 := v59 - v60\nj := v61", shape="rectangle", style="rounded"];
		subgraph cluster_main_loop3 {
			label="";
			color="blue";
			style="rounded";
			bgcolor="#FFFFFF00";
			main_9;
			main_7;
			main_3;
			main_8;
			main_4;
			main_10;
			main_6;
			main_5;
			subgraph cluster_main_loop5 {
				label="";
				color="green";
				style="rounded";
				bgcolor="#FFFFFF00";
				main_9;
				main_6;
				main_7;
				main_8;
				main_5;
			}
		}

		main_0 -> main_2;
		main_2 -> main_3;
		main_3 -> main_4 [label="T"];
		main_4 -> main_5;
		main_5 -> main_6 [label="T"];
		main_5 -> main_10 [label="F"];
		main_6 -> main_7 [label="T"];
		main_6 -> main_8 [label="F"];
		main_7 -> main_9;
		main_8 -> main_9;
		main_9 -> main_5;
		main_10 -> main_3;
	}
	subgraph cluster_f {
		label="f(X: float, Y: float, x: float, y: float, n: int)";
		rankdir="TB";
		f_0 [label="START", shape="rectangle", style="rounded"];
		f_2 [label="v0 := x\nv1 := x\nv2 := v0 * v1\nv3 := y\nv4 := y\nv5 := v3 * v4\nv6 := v2 + v5\nr := v6\nv8 := r\nv9 := 4\nv10 := v8 < v9\nif v10", shape="rectangle", style="rounded"];
		f_3 [label="v12 := n\nv13 := 100\nv14 := v12 < v13\nif v14", shape="rectangle", style="rounded"];
		f_4 [label="v15 := x\nv16 := x\nv17 := v15 * v16\nv18 := y\nv19 := y\nv20 := v18 * v19\nv21 := v17 - v20\nv22 := X\nv23 := v21 + v22\nx2 := v23\nv24 := 2\nv25 := x\nv26 := v24 * v25\nv27 := y\nv28 := v26 * v27\nv29 := Y\nv30 := v28 + v29\ny2 := v30\nv31 := 1\nv32 := X\nv33 := Y\nv34 := x2\nv35 := y2\nv36 := n\nv37 := 1\nv38 := v36 + v37\nv39 := f(v32, v33, v34, v35, v38)\nv40 := v31 + v39\nfv := v40\nv41 := fv\nreturn v41", shape="rectangle", style="rounded"];
		f_7 [label="v42 := 0\nreturn v42", shape="rectangle", style="rounded"];

		f_0 -> f_2;
		f_2 -> f_3 [label="T"];
		f_2 -> f_7 [label="F"];
		f_3 -> f_4 [label="T"];
		f_3 -> f_7 [label="F"];
	}
	subgraph cluster_pow10 {
		label="pow10(n: int, e: int)";
		rankdir="TB";
		pow10_0 [label="START", shape="rectangle", style="rounded"];
		pow10_2 [label="v1 := e\ni := v1", shape="rectangle", style="rounded"];
		pow10_3 [label="v2 := i\nv3 := 0\nv4 := v2 > v3\nif v4", shape="rectangle", style="rounded"];
		pow10_4 [label="v5 := n\nv6 := 10\nv7 := v5 * v6\nn := v7\nv8 := i\nv9 := 1\nv10 := v8 - v9\ni := v10", shape="rectangle", style="rounded"];
		pow10_5 [label="v11 := n\nreturn v11", shape="rectangle", style="rounded"];
		subgraph cluster_pow10_loop3 {
			label="";
			color="blue";
			style="rounded";
			bgcolor="#FFFFFF00";
			pow10_3;
			pow10_4;
		}

		pow10_0 -> pow10_2;
		pow10_2 -> pow10_3;
		pow10_3 -> pow10_4 [label="T"];
		pow10_3 -> pow10_5 [label="F"];
		pow10_4 -> pow10_3;
	}
}
