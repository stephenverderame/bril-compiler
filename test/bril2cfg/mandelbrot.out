digraph {
	subgraph cluster_main {
		label="main";
		rankdir="TB";
		main_0 [label="START", shape="ellipse"];
		main_2 [label="v12 := 1", shape="ellipse"];
		main_3 [label="space := v12", shape="ellipse"];
		main_4 [label="v13 := 8", shape="ellipse"];
		main_5 [label="value := v13", shape="ellipse"];
		main_6 [label="v14 := 0", shape="ellipse"];
		main_7 [label="v15 := 1", shape="ellipse"];
		main_8 [label="v16 := v14 - v15", shape="ellipse"];
		main_9 [label="neg_1 := v16", shape="ellipse"];
		main_10 [label="v17 := 0", shape="ellipse"];
		main_11 [label="v18 := 2", shape="ellipse"];
		main_12 [label="v19 := v17 - v18", shape="ellipse"];
		main_13 [label="neg_2 := v19", shape="ellipse"];
		main_14 [label="v21 := 1", shape="ellipse"];
		main_15 [label="j := v21", shape="ellipse"];
		main_16 [label="v22 := j", shape="ellipse"];
		main_17 [label="v23 := neg_1", shape="ellipse"];
		main_18 [label="v24 := v22 >= v23", shape="ellipse"];
		main_19 [label="if v24", shape="ellipse"];
		main_20 [label="v25 := 0", shape="ellipse"];
		main_21 [label="num := v25", shape="ellipse"];
		main_22 [label="v26 := 0", shape="ellipse"];
		main_23 [label="cnt := v26", shape="ellipse"];
		main_24 [label="v28 := neg_2", shape="ellipse"];
		main_25 [label="i := v28", shape="ellipse"];
		main_26 [label="v29 := i", shape="ellipse"];
		main_27 [label="v30 := 0.5", shape="ellipse"];
		main_28 [label="v31 := v29 <= v30", shape="ellipse"];
		main_29 [label="if v31", shape="ellipse"];
		main_30 [label="v32 := i", shape="ellipse"];
		main_31 [label="v33 := j", shape="ellipse"];
		main_32 [label="v34 := 0", shape="ellipse"];
		main_33 [label="v35 := 0", shape="ellipse"];
		main_34 [label="v36 := 0", shape="ellipse"];
		main_35 [label="x := f(v32, v33, v34, v35, v36)", shape="ellipse"];
		main_36 [label="x := x", shape="ellipse"];
		main_37 [label="v38 := x", shape="ellipse"];
		main_38 [label="v39 := 50", shape="ellipse"];
		main_39 [label="v40 := v38 < v39", shape="ellipse"];
		main_40 [label="if v40", shape="ellipse"];
		main_41 [label="v41 := space", shape="ellipse"];
		main_42 [label="v42 := cnt", shape="ellipse"];
		main_43 [label="v := pow10(v41, v42)", shape="ellipse"];
		main_44 [label="v := v", shape="ellipse"];
		main_45 [label="v43 := num", shape="ellipse"];
		main_46 [label="v44 := v", shape="ellipse"];
		main_47 [label="v45 := v43 + v44", shape="ellipse"];
		main_48 [label="num := v45", shape="ellipse"];
		main_50 [label="v46 := value", shape="ellipse"];
		main_51 [label="v47 := cnt", shape="ellipse"];
		main_52 [label="v := pow10(v46, v47)", shape="ellipse"];
		main_53 [label="v := v", shape="ellipse"];
		main_54 [label="v48 := num", shape="ellipse"];
		main_55 [label="v49 := v", shape="ellipse"];
		main_56 [label="v50 := v48 + v49", shape="ellipse"];
		main_57 [label="num := v50", shape="ellipse"];
		main_58 [label="v51 := cnt", shape="ellipse"];
		main_59 [label="v52 := 1", shape="ellipse"];
		main_60 [label="v53 := v51 + v52", shape="ellipse"];
		main_61 [label="cnt := v53", shape="ellipse"];
		main_62 [label="v54 := i", shape="ellipse"];
		main_63 [label="v55 := 0.225", shape="ellipse"];
		main_64 [label="v56 := v54 + v55", shape="ellipse"];
		main_65 [label="i := v56", shape="ellipse"];
		main_67 [label="v57 := num", shape="ellipse"];
		main_68 [label="print v57", shape="ellipse"];
		main_69 [label="v58 := 0", shape="ellipse"];
		main_70 [label="v59 := j", shape="ellipse"];
		main_71 [label="v60 := 0.015", shape="ellipse"];
		main_72 [label="v61 := v59 - v60", shape="ellipse"];
		main_73 [label="j := v61", shape="ellipse"];

		main_0 -> main_2;
		main_2 -> main_3;
		main_3 -> main_4;
		main_4 -> main_5;
		main_5 -> main_6;
		main_6 -> main_7;
		main_7 -> main_8;
		main_8 -> main_9;
		main_9 -> main_10;
		main_10 -> main_11;
		main_11 -> main_12;
		main_12 -> main_13;
		main_13 -> main_14;
		main_14 -> main_15;
		main_15 -> main_16;
		main_16 -> main_17;
		main_17 -> main_18;
		main_18 -> main_19;
		main_19 -> main_20 [label="T"];
		main_20 -> main_21;
		main_21 -> main_22;
		main_22 -> main_23;
		main_23 -> main_24;
		main_24 -> main_25;
		main_25 -> main_26;
		main_26 -> main_27;
		main_27 -> main_28;
		main_28 -> main_29;
		main_29 -> main_30 [label="T"];
		main_29 -> main_67 [label="F"];
		main_30 -> main_31;
		main_31 -> main_32;
		main_32 -> main_33;
		main_33 -> main_34;
		main_34 -> main_35;
		main_35 -> main_36;
		main_36 -> main_37;
		main_37 -> main_38;
		main_38 -> main_39;
		main_39 -> main_40;
		main_40 -> main_41 [label="T"];
		main_40 -> main_50 [label="F"];
		main_41 -> main_42;
		main_42 -> main_43;
		main_43 -> main_44;
		main_44 -> main_45;
		main_45 -> main_46;
		main_46 -> main_47;
		main_47 -> main_48;
		main_48 -> main_58;
		main_50 -> main_51;
		main_51 -> main_52;
		main_52 -> main_53;
		main_53 -> main_54;
		main_54 -> main_55;
		main_55 -> main_56;
		main_56 -> main_57;
		main_57 -> main_58;
		main_58 -> main_59;
		main_59 -> main_60;
		main_60 -> main_61;
		main_61 -> main_62;
		main_62 -> main_63;
		main_63 -> main_64;
		main_64 -> main_65;
		main_65 -> main_26;
		main_67 -> main_68;
		main_68 -> main_69;
		main_69 -> main_70;
		main_70 -> main_71;
		main_71 -> main_72;
		main_72 -> main_73;
		main_73 -> main_16;
	}
	subgraph cluster_f {
		label="f";
		rankdir="TB";
		f_0 [label="START", shape="ellipse"];
		f_2 [label="v0 := x", shape="ellipse"];
		f_3 [label="v1 := x", shape="ellipse"];
		f_4 [label="v2 := v0 * v1", shape="ellipse"];
		f_5 [label="v3 := y", shape="ellipse"];
		f_6 [label="v4 := y", shape="ellipse"];
		f_7 [label="v5 := v3 * v4", shape="ellipse"];
		f_8 [label="v6 := v2 + v5", shape="ellipse"];
		f_9 [label="r := v6", shape="ellipse"];
		f_10 [label="v8 := r", shape="ellipse"];
		f_11 [label="v9 := 4", shape="ellipse"];
		f_12 [label="v10 := v8 < v9", shape="ellipse"];
		f_13 [label="if v10", shape="ellipse"];
		f_14 [label="v12 := n", shape="ellipse"];
		f_15 [label="v13 := 100", shape="ellipse"];
		f_16 [label="v14 := v12 < v13", shape="ellipse"];
		f_17 [label="if v14", shape="ellipse"];
		f_18 [label="v15 := x", shape="ellipse"];
		f_19 [label="v16 := x", shape="ellipse"];
		f_20 [label="v17 := v15 * v16", shape="ellipse"];
		f_21 [label="v18 := y", shape="ellipse"];
		f_22 [label="v19 := y", shape="ellipse"];
		f_23 [label="v20 := v18 * v19", shape="ellipse"];
		f_24 [label="v21 := v17 - v20", shape="ellipse"];
		f_25 [label="v22 := X", shape="ellipse"];
		f_26 [label="v23 := v21 + v22", shape="ellipse"];
		f_27 [label="x2 := v23", shape="ellipse"];
		f_28 [label="v24 := 2", shape="ellipse"];
		f_29 [label="v25 := x", shape="ellipse"];
		f_30 [label="v26 := v24 * v25", shape="ellipse"];
		f_31 [label="v27 := y", shape="ellipse"];
		f_32 [label="v28 := v26 * v27", shape="ellipse"];
		f_33 [label="v29 := Y", shape="ellipse"];
		f_34 [label="v30 := v28 + v29", shape="ellipse"];
		f_35 [label="y2 := v30", shape="ellipse"];
		f_36 [label="v31 := 1", shape="ellipse"];
		f_37 [label="v32 := X", shape="ellipse"];
		f_38 [label="v33 := Y", shape="ellipse"];
		f_39 [label="v34 := x2", shape="ellipse"];
		f_40 [label="v35 := y2", shape="ellipse"];
		f_41 [label="v36 := n", shape="ellipse"];
		f_42 [label="v37 := 1", shape="ellipse"];
		f_43 [label="v38 := v36 + v37", shape="ellipse"];
		f_44 [label="v39 := f(v32, v33, v34, v35, v38)", shape="ellipse"];
		f_45 [label="v40 := v31 + v39", shape="ellipse"];
		f_46 [label="fv := v40", shape="ellipse"];
		f_47 [label="v41 := fv", shape="ellipse"];
		f_48 [label="return v41", shape="ellipse"];
		f_51 [label="v42 := 0", shape="ellipse"];
		f_52 [label="return v42", shape="ellipse"];

		f_0 -> f_2;
		f_2 -> f_3;
		f_3 -> f_4;
		f_4 -> f_5;
		f_5 -> f_6;
		f_6 -> f_7;
		f_7 -> f_8;
		f_8 -> f_9;
		f_9 -> f_10;
		f_10 -> f_11;
		f_11 -> f_12;
		f_12 -> f_13;
		f_13 -> f_14 [label="T"];
		f_13 -> f_51 [label="F"];
		f_14 -> f_15;
		f_15 -> f_16;
		f_16 -> f_17;
		f_17 -> f_18 [label="T"];
		f_17 -> f_51 [label="F"];
		f_18 -> f_19;
		f_19 -> f_20;
		f_20 -> f_21;
		f_21 -> f_22;
		f_22 -> f_23;
		f_23 -> f_24;
		f_24 -> f_25;
		f_25 -> f_26;
		f_26 -> f_27;
		f_27 -> f_28;
		f_28 -> f_29;
		f_29 -> f_30;
		f_30 -> f_31;
		f_31 -> f_32;
		f_32 -> f_33;
		f_33 -> f_34;
		f_34 -> f_35;
		f_35 -> f_36;
		f_36 -> f_37;
		f_37 -> f_38;
		f_38 -> f_39;
		f_39 -> f_40;
		f_40 -> f_41;
		f_41 -> f_42;
		f_42 -> f_43;
		f_43 -> f_44;
		f_44 -> f_45;
		f_45 -> f_46;
		f_46 -> f_47;
		f_47 -> f_48;
		f_51 -> f_52;
	}
	subgraph cluster_pow10 {
		label="pow10";
		rankdir="TB";
		pow10_0 [label="START", shape="ellipse"];
		pow10_2 [label="v1 := e", shape="ellipse"];
		pow10_3 [label="i := v1", shape="ellipse"];
		pow10_4 [label="v2 := i", shape="ellipse"];
		pow10_5 [label="v3 := 0", shape="ellipse"];
		pow10_6 [label="v4 := v2 > v3", shape="ellipse"];
		pow10_7 [label="if v4", shape="ellipse"];
		pow10_8 [label="v5 := n", shape="ellipse"];
		pow10_9 [label="v6 := 10", shape="ellipse"];
		pow10_10 [label="v7 := v5 * v6", shape="ellipse"];
		pow10_11 [label="n := v7", shape="ellipse"];
		pow10_12 [label="v8 := i", shape="ellipse"];
		pow10_13 [label="v9 := 1", shape="ellipse"];
		pow10_14 [label="v10 := v8 - v9", shape="ellipse"];
		pow10_15 [label="i := v10", shape="ellipse"];
		pow10_17 [label="v11 := n", shape="ellipse"];
		pow10_18 [label="return v11", shape="ellipse"];

		pow10_0 -> pow10_2;
		pow10_2 -> pow10_3;
		pow10_3 -> pow10_4;
		pow10_4 -> pow10_5;
		pow10_5 -> pow10_6;
		pow10_6 -> pow10_7;
		pow10_7 -> pow10_8 [label="T"];
		pow10_7 -> pow10_17 [label="F"];
		pow10_8 -> pow10_9;
		pow10_9 -> pow10_10;
		pow10_10 -> pow10_11;
		pow10_11 -> pow10_12;
		pow10_12 -> pow10_13;
		pow10_13 -> pow10_14;
		pow10_14 -> pow10_15;
		pow10_15 -> pow10_4;
		pow10_17 -> pow10_18;
	}
}
